Fixed AM_Q Modulation, added x10 protocol

grugnoymeme opened the pull request:
# What's new

- Fixed AM_q Modulation from @haisenteck -  https://github.com/haisenteck/flipperzero-Haisenteck
- addes x10 protocol for subghz from @jamisonderek

Sil333033 commented:
Protocol looks great thanks. Did you confirm that it worked? Can i see it in action?

What is fixed with the AM_Q modulation? Did heisenburg made a magic spell so the flipper doesn't freeze with external radio?

grugnoymeme commented:
https://github.com/jamisonderek/flipper-zero-tutorials/tree/main/subghz/samples/x10

here u'll find lots of x10 signals sample...

Willy-JL commented:
and even then @haisenteck, how you implemented the AM650 disguised as AM_Q in your "firmware" is wrong too. if you receive with your "AM_Q" and try to save the recorded file, then you will not be able to load it again because the modulation is implemented only for saving, not loading.

flipper firmware development is *clearly* too complicated for you, go back to airsoft.

haisenteck commented:
> > > Protocol looks great thanks. Did you confirm that it worked? Can i see it in action?
> > > What is fixed with the AM_Q modulation? Did heisenburg made a magic spell so the flipper doesn't freeze with external radio?
> > 
> > 
> > resolved the fact that it was literally with missing data in the register, that the configuration was physically incorrect and out of specification of the cc1101, found the value for which it made the hop jump between frequencies and corrected the data rate to resolve the fact that with the fixes had gone out of specification. Soon I will also solve HONDA1 and HONDA2 and TPMS which every now and then still plays tricks by misreading the data by setting everything to 0 on its own.
> 
> # no @haisenteck, you are a FRAUD.
> let me explain:
> 
> * take the AM650 preset data from `lib/subghz/devices/cc1101_configs.c` in OFW and resolve the defined names to binary values, you get:
> * `Real AM650:`
>   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> * take the AM_Q preset data from `lib/subghz/devices/cc1101_configs.c` in Haisenteck and resolve the defined names to binary values, you get:
> * `Haisenteck "AM_Q":`
>   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 00 C0 00 00 00 00 00 00 00`
> * take the AM_Q preset data from Derek Jamiason's actual video where he _made_ AM_Q, you get:
> * `Real AM_Q:`
>   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 22 10 1C 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> 
> let me break this down. if you watch derek's video you will know that the important values are in registers `11` and `10`. the real `AM650` has `11 32` and `10 17`. the real `AM_Q` has `11 22` and `10 1C`. other than these 2 values, everything else is the same. now look at @haisenteck's `"AM_Q"`, it has `11 32` and `10 17` EXACTLY LIKE `AM650`. the only values that make the difference between `AM650` and `AM_Q` are the same as `AM650` in @haisenteck's `"AM_Q"`. for some reason he felt the need to add some 2 additional `00` bytes to pad out the rest of the preset, in places where they conveniently do nothing at all, since it is outside the `load reg` section as you can see if you compare his implementation to the very well documented one in `lib/subghz/devices/cc1101_configs.c`
> 
> so in summary, NO @haisenteck DID NOT FIX AM_Q. he made an AM650 clone with 2 additional empty bytes that do NOTHING. the only AM_Q working version is from Derek Jamiason and as he said himself, it has issues with external modules due to the elevated data rate. stop making up bullshit and stop being a skid @haisenteck. go do something else if you want attention, because in this space people will find out your wrongdoings and call you out for it.

first, ok, not solved the variable that i apply in the same position off all the preset, but i correct the other data from "SmartRF Studio 7" of Texas Instrument for a work. i do not verify if are the same of another preset.

second: the missing 00 data is important because since the configuration register of the CC1101 is positional in the configurations, the power parameter was passed not as C0 (exit power +10) but as 00 (exit power 0). now that I notice it, you also pointed out to me that when I did the last upload to github the transposition errors remained (which also don't even make everything work in that way as too many values are passed).

third: It's a fix and my implementation that I'm doing, it's not finished or fixed overall. I wasn't the one doing the pull or going around saying it's finished or fully functional.

fourth: I don't want attention and I've never even cared about spreading this stuff. since I'm developing for myself and I have no interest in making all this mess around. the fact that there are people who take stuff and send it around on their own business is not within my competence since I have never sent you anything and indeed, as I told you, if you are interested in something, take it even if it is open source with the usual license or as with all things on this planet... just ignore it.

finally: if you had ever read what I wrote myself about the Grocery Store app... you would know that it was my first fap, made to test and see to what extent I was able to attempt to create a fap from scratch . lasted 2 weeks, removed after a time which was useless as I had even forgotten about it after I deleted the things necessary to make it work for something else in favor of yours with as many credits for having taken it.

that said: I'm going back to minding my own business with my firmware and renewing the issue, if you're interested in something, take it and say thank you like Korai also does, for the rest if you're not interested, there's no point in attacking if they pass you stuff that you don't like. I never said I was a god or that I knew how to do everything, I enjoy myself and work on things that interest me that apparently you don't like.

Good day.

grugnoymeme commented:
> > > > Protocol looks great thanks. Did you confirm that it worked? Can i see it in action?
> > > > What is fixed with the AM_Q modulation? Did heisenburg made a magic spell so the flipper doesn't freeze with external radio?
> > > 
> > > 
> > > resolved the fact that it was literally with missing data in the register, that the configuration was physically incorrect and out of specification of the cc1101, found the value for which it made the hop jump between frequencies and corrected the data rate to resolve the fact that with the fixes had gone out of specification. Soon I will also solve HONDA1 and HONDA2 and TPMS which every now and then still plays tricks by misreading the data by setting everything to 0 on its own.
> > 
> > # no @haisenteck, you are a FRAUD.
> > let me explain:
> > 
> > * take the AM650 preset data from `lib/subghz/devices/cc1101_configs.c` in OFW and resolve the defined names to binary values, you get:
> > * `Real AM650:`
> >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> > * take the AM_Q preset data from `lib/subghz/devices/cc1101_configs.c` in Haisenteck and resolve the defined names to binary values, you get:
> > * `Haisenteck "AM_Q":`
> >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 00 C0 00 00 00 00 00 00 00`
> > * take the AM_Q preset data from Derek Jamiason's actual video where he _made_ AM_Q, you get:
> > * `Real AM_Q:`
> >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 22 10 1C 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> > 
> > let me break this down. if you watch derek's video you will know that the important values are in registers `11` and `10`. the real `AM650` has `11 32` and `10 17`. the real `AM_Q` has `11 22` and `10 1C`. other than these 2 values, everything else is the same. now look at @haisenteck's `"AM_Q"`, it has `11 32` and `10 17` EXACTLY LIKE `AM650`. the only values that make the difference between `AM650` and `AM_Q` are the same as `AM650` in @haisenteck's `"AM_Q"`. for some reason he felt the need to add some 2 additional `00` bytes to pad out the rest of the preset, in places where they conveniently do nothing at all, since it is outside the `load reg` section as you can see if you compare his implementation to the very well documented one in `lib/subghz/devices/cc1101_configs.c`
> > 
> > so in summary, NO @haisenteck DID NOT FIX AM_Q. he made an AM650 clone with 2 additional empty bytes that do NOTHING. the only AM_Q working version is from Derek Jamiason and as he said himself, it has issues with external modules due to the elevated data rate. stop making up bullshit and stop being a skid @haisenteck. go do something else if you want attention, because in this space people will find out your wrongdoings and call you out for it.
> 
> first, ok, not solved the variable that i apply in the same position off all the preset, but i correct the other data from "SmartRF Studio 7" of Texas Instrument for a work. i do not verify if are the same of another preset.
> 
> second: the missing 00 data is important because since the configuration register of the CC1101 is positional in the configurations, the power parameter was passed not as C0 (exit power +10) but as 00 (exit power 0). now that I notice it, you also pointed out to me that when I did the last upload to github the transposition errors remained (which also don't even make everything work in that way as too many values are passed).
> 
> third: It's a fix and my implementation that I'm doing, it's not finished or fixed overall. I wasn't the one doing the pull or going around saying it's finished or fully functional.
> 
> fourth: I don't want attention and I've never even cared about spreading this stuff. since I'm developing for myself and I have no interest in making all this mess around. the fact that there are people who take stuff and send it around on their own business is not within my competence since I have never sent you anything and indeed, as I told you, if you are interested in something, take it even if it is open source with the usual license or as with all things on this planet... just ignore it.
> 
> finally: if you had ever read what I wrote myself about the Grocery Store app... you would know that it was my first fap, made to test and see to what extent I was able to attempt to create a fap from scratch . lasted 2 weeks, removed after a time which was useless as I had even forgotten about it after I deleted the things necessary to make it work for something else in favor of yours with as many credits for having taken it.
> 
> that said: I'm going back to minding my own business with my firmware and renewing the issue, if you're interested in something, take it and say thank you like Korai also does, for the rest if you're not interested, there's no point in attacking if they pass you stuff that you don't like. I never said I was a god or that I knew how to do everything, I enjoy myself and work on things that interest me that apparently you don't like.
> 
> Good day.

The third poi Is fake! You didnt want to pull nothing BC "no time to lose" i asked you mire than One time if It was completely fixed... So don't PULL the fault only tò me... I asked Sorry for the impulsivity BC i don't know subghz so good and i pulled something fake... (And you said "exactely" when i said tò you: i am not Ur pigeon, i make the PR JUST TO MAKE FASTER) PLEASE....

grugnoymeme commented:
@haisenteck you asked me in public tò PR SAYING THAT I TOOK IT FROM U...

haisenteck commented:
> @haisenteck you asked me in public tò PR SAYING THAT I TOOK IT FROM U...



> > > > > Protocol looks great thanks. Did you confirm that it worked? Can i see it in action?
> > > > > What is fixed with the AM_Q modulation? Did heisenburg made a magic spell so the flipper doesn't freeze with external radio?
> > > > 
> > > > 
> > > > resolved the fact that it was literally with missing data in the register, that the configuration was physically incorrect and out of specification of the cc1101, found the value for which it made the hop jump between frequencies and corrected the data rate to resolve the fact that with the fixes had gone out of specification. Soon I will also solve HONDA1 and HONDA2 and TPMS which every now and then still plays tricks by misreading the data by setting everything to 0 on its own.
> > > 
> > > 
> > > # no @haisenteck, you are a FRAUD.
> > > let me explain:
> > > 
> > > * take the AM650 preset data from `lib/subghz/devices/cc1101_configs.c` in OFW and resolve the defined names to binary values, you get:
> > > * `Real AM650:`
> > >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> > > * take the AM_Q preset data from `lib/subghz/devices/cc1101_configs.c` in Haisenteck and resolve the defined names to binary values, you get:
> > > * `Haisenteck "AM_Q":`
> > >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 32 10 17 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 00 C0 00 00 00 00 00 00 00`
> > > * take the AM_Q preset data from Derek Jamiason's actual video where he _made_ AM_Q, you get:
> > > * `Real AM_Q:`
> > >   `02 0D 03 07 08 32 0B 06 14 00 13 00 12 30 11 22 10 1C 18 18 19 18 1D 91 1C 00 1B 07 20 FB 22 11 21 B6 00 00 00 C0 00 00 00 00 00 00`
> > > 
> > > let me break this down. if you watch derek's video you will know that the important values are in registers `11` and `10`. the real `AM650` has `11 32` and `10 17`. the real `AM_Q` has `11 22` and `10 1C`. other than these 2 values, everything else is the same. now look at @haisenteck's `"AM_Q"`, it has `11 32` and `10 17` EXACTLY LIKE `AM650`. the only values that make the difference between `AM650` and `AM_Q` are the same as `AM650` in @haisenteck's `"AM_Q"`. for some reason he felt the need to add some 2 additional `00` bytes to pad out the rest of the preset, in places where they conveniently do nothing at all, since it is outside the `load reg` section as you can see if you compare his implementation to the very well documented one in `lib/subghz/devices/cc1101_configs.c`
> > > so in summary, NO @haisenteck DID NOT FIX AM_Q. he made an AM650 clone with 2 additional empty bytes that do NOTHING. the only AM_Q working version is from Derek Jamiason and as he said himself, it has issues with external modules due to the elevated data rate. stop making up bullshit and stop being a skid @haisenteck. go do something else if you want attention, because in this space people will find out your wrongdoings and call you out for it.
> > 
> > 
> > first, ok, not solved the variable that i apply in the same position off all the preset, but i correct the other data from "SmartRF Studio 7" of Texas Instrument for a work. i do not verify if are the same of another preset.
> > second: the missing 00 data is important because since the configuration register of the CC1101 is positional in the configurations, the power parameter was passed not as C0 (exit power +10) but as 00 (exit power 0). now that I notice it, you also pointed out to me that when I did the last upload to github the transposition errors remained (which also don't even make everything work in that way as too many values are passed).
> > third: It's a fix and my implementation that I'm doing, it's not finished or fixed overall. I wasn't the one doing the pull or going around saying it's finished or fully functional.
> > fourth: I don't want attention and I've never even cared about spreading this stuff. since I'm developing for myself and I have no interest in making all this mess around. the fact that there are people who take stuff and send it around on their own business is not within my competence since I have never sent you anything and indeed, as I told you, if you are interested in something, take it even if it is open source with the usual license or as with all things on this planet... just ignore it.
> > finally: if you had ever read what I wrote myself about the Grocery Store app... you would know that it was my first fap, made to test and see to what extent I was able to attempt to create a fap from scratch . lasted 2 weeks, removed after a time which was useless as I had even forgotten about it after I deleted the things necessary to make it work for something else in favor of yours with as many credits for having taken it.
> > that said: I'm going back to minding my own business with my firmware and renewing the issue, if you're interested in something, take it and say thank you like Korai also does, for the rest if you're not interested, there's no point in attacking if they pass you stuff that you don't like. I never said I was a god or that I knew how to do everything, I enjoy myself and work on things that interest me that apparently you don't like.
> > Good day.
> 
> The third poi Is fake! You didnt want to pull nothing BC "no time to lose" i asked you mire than One time if It was completely fixed... So don't PULL the fault only tò me... I asked Sorry for the impulsivity BC i don't know subghz so good and i pulled something fake... (And you said "exactely" when i said tò you: i am not Ur pigeon, i make the PR JUST TO MAKE FASTER) PLEASE....

![image](https://github.com/Flipper-XFW/Xtreme-Firmware/assets/142255985/e54298b1-1279-4afe-a9a6-b91c8aaa4cec)
te lo scrivo in italiano, tanto non serve che te lo traduco: non ho mai scritto di aver finito. mi hai chiesto solo se funzionava ed ho riposto esattamente come qua, ho messo gli screen del test in corso.

per il discorso di evitarti di fare il piccione viaggiatore con i messaggi è stato proprio perchè tu non sei l'autore di certe cose, quindi anche se ti fanno domande o ti danno risposte devi riportare. se vogliono qualcosa sanno come si scrive, come io se ho domande per loro posso sempre scrivere. ti ho solo tolto di mezzo la cosa di fare avanti e indietro a riportare botta e risposta e sono venuto io qui a rispondere (che per altro potevi anche dirmi prima che volevi caricare roba in giro, almeno ti rispondevo già li che non era completo o finito ma fai sempre di testa tua senza chiedere...)

haisenteck commented:
> @haisenteck you asked me in public tò PR SAYING THAT I TOOK IT FROM U...

certo, se no facevi come per la lettura NFC, che hai fatto il pull senza riportare le fonti ma come se il lavoro lo avessi fatto tu per i fatti tuoi e per altro non ti ho ripreso io sulla comunity, ma un altro utente che ti ha già detto il tutto. io mi sono solo limitato a chiederti di riportare le fonti, che il lavoro piaccia o meno l'opensource funziona proprio così.

Willy-JL commented:
> > @haisenteck you asked me in public tò PR SAYING THAT I TOOK IT FROM U...
> 
> certo, se no facevi come per la lettura NFC, che hai fatto il pull senza riportare le fonti ma come se il lavoro lo avessi fatto tu per i fatti tuoi e per altro non ti ho ripreso io sulla comunity, ma un altro utente che ti ha già detto il tutto. io mi sono solo limitato a chiederti di riportare le fonti, che il lavoro piaccia o meno l'opensource funziona proprio così.

In termini di AM_Q qui di lavoro non ce n'è  nemmeno un granello di polvere. L'unico motivo per cui AM_Q esiste è la data rate superiore, che si imposta tramite i registri 0x11 e 0x10. Nel tuo "AM_Q" quei due registri hanno gli stessi valori di AM650, quindi qui di AM_Q non c'è nemmeno l'ombra. E per quando riguarda gli altri due zeri, ti sbagli. Tu non sai una minchia di niente di subghz e ti incinti un paio di minchiate per sembrare uno sviluppatore quando in realtà non sai nemmeno usare git merge. Visto che a sto punto il tuo "AM_Q" è esattamente uguale a AM650, se quei due 00 byte mancanti fossero un problema, sarebbero un problema per tutti su AM650z. Ma se guardassi le specifiche di Texasinstruments vedresti che in quella posizione I byte 00 non cambiano una cicca di tabacco. Quindi muto, ritorna nel tuo triste angolino a fare il lecchino su telegram.
